src/components/screens/login/login.js

```diff
@@ -17,164 +14,115 @@ import DiscreetModeToggle from '@shared/discreetModeToggle';
-      isValid: false,
-      passphrase: '',
-      network: loginNetwork.name,
-      address,
-      validationError: false,
-    };
-
-    this.secondIteration = false;
-
-    this.networks = getNetworksList();
+const RegisterTitle = ({ t }) => (
+  <div className={`${styles.titleHolder} ${grid['col-xs-10']}`}>
+    <h1>
+      {t('Sign in with a Passphrase')}
+    </h1>
+    <p>
+      {t('Don’t have a Lisk account yet? ')}
+      <Link className={styles.link} to={routes.register.path}>
+        {t('Create it now')}
+      </Link>
+    </p>
+  </div>
+);
+
+/**
+ * Get referer address  and redirect to, after signing in
+ * @param {object} history - The history object from react-router
+ */
+const redirectToReferrer = (history) => {
+  const { referrer, ...restParams } = parseSearchParams(history.location.search);
+  const route = referrer ? `${referrer}${stringifySearchParams(restParams)}` : routes.dashboard.path;
+
+  history.replace(route);
+};
+
+const Login = ({
+  t, settings, network, history, account, login,
+}) => {
+  const [passphrase, setPass] = useState({ value: '', isValid: false });
+  const setPassphrase = (value, error) => {
+    setPass({
+      value,
+      isValid: !error,
+    });
+  };
 
-    this.checkPassphrase = this.checkPassphrase.bind(this);
-    this.onFormSubmit = this.onFormSubmit.bind(this);
-    this.onLoginSubmission = this.onLoginSubmission.bind(this);
-  }
+  const onFormSubmit = (e) => {
+    e.preventDefault();
+    Piwik.trackingEvent('Login', 'button', 'Login submission');
+    login({ passphrase: passphrase.value });
+  };
 
-  async componentDidMount() {
+  useEffect(() => {
     // istanbul ignore else
-    if (!this.props.settings.areTermsOfUseAccepted && this.props.network.networks?.LSK) {
-      this.props.history.push(routes.termsOfUse.path);
+    if (!settings.areTermsOfUseAccepted && network.networks?.LSK) {
+      history.push(routes.termsOfUse.path);
     }
 
     i18next.on('languageChanged', getNetworksList);
-  }
+  }, []);
 
-  componentDidUpdate() {
-    if (this.props.account?.summary?.address) {
-      this.redirectToReferrer();
+  useEffect(() => {
+    if (account?.summary?.address) {
+      redirectToReferrer(history);
     }
-  }
-
-  getReferrerRoute() {
-    const { referrer, ...restParams } = parseSearchParams(this.props.history.location.search);
-    const route = referrer ? `${referrer}${stringifySearchParams(restParams)}` : routes.dashboard.path;
-    return route;
-  }
-
-  redirectToReferrer() {
-    this.props.history.replace(this.getReferrerRoute());
-  }
-
-  alreadyLoggedWithThisAddress(address, prevNetwork) {
-    const { account, network, settings: { token: { active } } } = this.props;
-    return account
-      && network
-      && account.summary?.address === address
-      && network.name === prevNetwork.name
-      && network.networks[active].nodeUrl === prevNetwork.networks[active].nodeUrl;
-  }
+  }, [account?.summary?.address]);
+
+  return (
+    <>
```

Comment:
whats the purpose of `<>` tag?

Reply:
React component must return a single child. In case you need to return multiple, you can wrap them in a temporary react component called a Fragment. If you're interested, [here](https://reactjs.org/docs/fragments.html) you can read more about it.